cpp/src/arrow/testing/gtest_util.h

```diff
@@ -345,4 +346,70 @@ void AssertSortedEquals(std::vector<T> u, std::vector<T> v) {
   ASSERT_EQ(u, v);
 }
 
+// Mixin used to generate trivial Array/RecordBatch/RecordBatchReader
+template <typename T>
+static inline std::shared_ptr<Array> GetArray(int64_t size,
+                                              std::shared_ptr<DataType> type,
+                                              T value = 0) {
+  using BuilderType = typename CTypeTraits<T>::BuilderType;
+  auto builder_fn = [](BuilderType* builder) { builder->UnsafeAppend(T(0)); };
+  ASSERT_OK_AND_ASSIGN(auto array, ArrayFromBuilderVisitor(type, size, builder_fn));
+  return array;
+}
```

Comment:
Personally I would implement `Zeros<T>(size)` to go alongside the current functions in testing/random.h